lib/std/json.zig

```diff
@@ -1504,7 +1504,15 @@ fn parseInternal(comptime T: type, token: Token, tokens: *TokenStream, options:
                                         parseFree(field.field_type, @field(r, field.name), options);
                                     }
                                 }
-                                @field(r, field.name) = try parse(field.field_type, tokens, options);
+                                if (field.is_comptime) {
+                                    const value = try parse(field.field_type, tokens, options);
+                                    defer parseFree(field.field_type, value, options);
+                                    if (value != @field(r, field.name)) {
```

Comment:
Possible future work: use other equality comparison here, e.g. might need to investigate `std.meta.eql`

Reply:
I added a low-effort version of this now to enable @marler8997